### ISSUE TITLE
Dialogs should have input field labels

### ISSUE THREAD

**Issue by blink1073** [jupyterlab/jupyterlab#1311]
cf https://github.com/jupyterlab/jupyterlab/pull/1305#issuecomment-263397010

**Comment by annabu** [comment]
Added reading: https://uxdesign.cc/alternatives-to-placeholder-text-13f430abc56f
